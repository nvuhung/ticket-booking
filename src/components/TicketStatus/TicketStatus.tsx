import React from 'react';
import styled from 'styled-components';

import { theme } from '../../styles/theme';
import { TicketStatuses } from '../../utils/enums';

const TicketStatus: React.FC = () => (
  <Container>
    <StatusItem>
      <StatusColor color={theme.colors.ticketStatus[TicketStatuses.Booked]} />
      <Text>Đã đặt</Text>
    </StatusItem>

    <StatusItem>
      <StatusColor color={theme.colors.ticketStatus[TicketStatuses.Selecting]} />
      <Text>Đang chọn</Text>
    </StatusItem>
  </Container>
);

const Container = styled.div``;

const StatusItem = styled.div`
  display: flex;
  align-items: center;
  margin-top: 10px;
`;

const StatusColor = styled.div<{ color: string }>`
  width: 18px;
  height: 18px;
  border-radius: 50%;
  background-color: ${props => props.color};
  margin-right: 10px;
`;

const Text = styled.span`
  color: ${theme.colors.white};
`;

export default React.memo(TicketStatus);
