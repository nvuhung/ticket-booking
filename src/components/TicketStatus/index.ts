import TicketStatus from './TicketStatus';

export default TicketStatus;
