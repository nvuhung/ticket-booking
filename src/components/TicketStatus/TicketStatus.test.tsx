import React from 'react';
import 'jest-styled-components';

import TicketStatus from './TicketStatus';
import { mount, shallow } from '../../test/enzyme';

describe('TicketStatus', () => {
  let mountedComp: any;

  const wrapper = (bShallow: boolean) => {
    if (!mountedComp) {
      mountedComp = bShallow ? shallow(<TicketStatus />) : mount(<TicketStatus />);
    }
    return mountedComp;
  };

  it('renders without crashing', () => {
    const comp = wrapper(false);
    expect(comp).toMatchSnapshot();
  });
});
