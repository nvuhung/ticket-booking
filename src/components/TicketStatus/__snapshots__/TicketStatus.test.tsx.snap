// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketStatus renders without crashing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 10px;
}

.c1 {
  width: 18px;
  height: 18px;
  border-radius: 50%;
  background-color: #242424;
  margin-right: 10px;
}

.c3 {
  width: 18px;
  height: 18px;
  border-radius: 50%;
  background-color: #fff;
  margin-right: 10px;
}

.c2 {
  color: #fff;
}

<Memo(TicketStatus)>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "bDWFJH",
            "rules": Array [],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className=""
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c0",
                  "rules": Array [
                    "
  display: flex;
  align-items: center;
  margin-top: 10px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c0"
            >
              <styled.div
                color="#242424"
              >
                <StyledComponent
                  color="#242424"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-htpNat",
                        "isStatic": false,
                        "lastClassName": "c3",
                        "rules": Array [
                          "
  width: 18px;
  height: 18px;
  border-radius: 50%;
  background-color: ",
                          [Function],
                          ";
  margin-right: 10px;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-htpNat",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="c1"
                    color="#242424"
                  />
                </StyledComponent>
              </styled.div>
              <styled.span>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bxivhb",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "
  color: ",
                          "#fff",
                          ";
",
                        ],
                      },
                      "displayName": "styled.span",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bxivhb",
                      "target": "span",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <span
                    className="c2"
                  >
                    Đã đặt
                  </span>
                </StyledComponent>
              </styled.span>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c0",
                  "rules": Array [
                    "
  display: flex;
  align-items: center;
  margin-top: 10px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c0"
            >
              <styled.div
                color="#fff"
              >
                <StyledComponent
                  color="#fff"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-htpNat",
                        "isStatic": false,
                        "lastClassName": "c3",
                        "rules": Array [
                          "
  width: 18px;
  height: 18px;
  border-radius: 50%;
  background-color: ",
                          [Function],
                          ";
  margin-right: 10px;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-htpNat",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="c3"
                    color="#fff"
                  />
                </StyledComponent>
              </styled.div>
              <styled.span>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bxivhb",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "
  color: ",
                          "#fff",
                          ";
",
                        ],
                      },
                      "displayName": "styled.span",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bxivhb",
                      "target": "span",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <span
                    className="c2"
                  >
                    Đang chọn
                  </span>
                </StyledComponent>
              </styled.span>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Memo(TicketStatus)>
`;
